llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -409,6 +435,10 @@ template <typename T>
 void ChangeReporter<T>::handleInvalidatedPass(StringRef PassID) {
   assert(!BeforeStack.empty() && "Unexpected empty stack encountered.");
 
+  // Prepare to process the next MIR.
+  if (PassID == FreeMachineFunctionPass::name())
```

Comment:
can you explain why this is necessary?

I think in general I'm not understanding the intent with "InitialIR" in regards to MIR. Previously we'd always start with the entire module as the initial IR right? Why are we handling MachineFunctions differently?

Reply:
`InitialIR` is used to determine when to call `handleInitialIR`, which is typically used to print to the user "Initial IR: ...". I don't think we need that for MIR, except in the case where we start with MIR (mostly just tests). We can just let before/after pass handle MIR?